src_c/line.c

```diff
@@ -125,270 +59,194 @@ pg_line_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 }
 
 static void
-pg_line_dealloc(pgLineObject *self)
-{
+pg_line_dealloc(pgLineObject *self) {
     if (self->weakreflist != NULL) {
         PyObject_ClearWeakRefs((PyObject *)self);
     }
 
     Py_TYPE(self)->tp_free((PyObject *)self);
 }
 
-static pgLineBase *
-pgLine_FromObject(PyObject *obj, pgLineBase *temp)
-{
+static int
+pgLine_FromObject(PyObject *obj, pgLineBase* out) {
     Py_ssize_t length;
-    PyObject *fseq = NULL;
+    PyObject* fseq = NULL;
 
     if (pgLine_Check(obj)) {
-        return &((pgLineObject *)obj)->line;
+        *out = ((pgLineObject *)obj)->line;
```

Comment:
In my opinion you should use a memcpy here, in that way you won't have to copy each attribute when you exit the function, kinda like the following:
```C
memcpy(out, &((pgLineObject *)obj)->line, sizeof(double)*4);
return 1;
```
the __init__ function would then become(for example)

```C
static int
pg_line_init(pgLineObject *self, PyObject *args, PyObject *kwds) {
    pgLineBase argrect;

    if (!pgLine_FromObject(args, &(self->line))) {
        PyErr_SetString(PyExc_TypeError, "Argument must be rect style object");
        return -1;
    }
    return 0;
}
```